src/Foundatio/Storage/FolderFileStorage.cs

```diff
@@ -77,22 +81,32 @@ public class FolderFileStorage : IFileStorage {
             path = path.NormalizePath();
 
             string file = Path.Combine(Folder, path);
-            string directory = Path.GetDirectoryName(file);
-            if (directory != null)
-                Directory.CreateDirectory(directory);
 
             try {
-                using (var fileStream = File.Create(file)) {
+                using (var fileStream = CreateFileStream(file)) {
                     if (stream.CanSeek)
                         stream.Seek(0, SeekOrigin.Begin);
 
                     stream.CopyTo(fileStream);
                 }
-            } catch (Exception) {
+            } catch (Exception ex) {
+                _logger.LogError(ex, "Error trying to save file '{path}'.", path);
```

Comment:
Log messages are automatically quoted in structured logging:

`_logger.LogError(ex, "Error trying to save file: {Path}", path);`